### PR TITLE
update dimensions

### LINKED ISSUE #5395: Update SL Dimensions SCD section to comply with the validation rules
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/docs/build/dimensions

### What part(s) of the page would you like to see updated?

Current'y it's not very clear how to properly use SCD dimensions:  
1. In the top table in `type` field we say "There are three types", but in YML there are only 2 options possible, which is fine, but quite confusing
2. In all dimension examples it's shown `metric_time` as a dimension, however it's a reserved word and such dimension is not allowed
3. `Note: The SCD dimensions table must have valid_to and valid_from columns.` -- it's not clear if these columns are really physically should exist or just configured via `validity_params : is_start / is_end`

### Additional information

_No response_

### LINKED ISSUE #5396: Improve SL metrics examples
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/docs/build/metrics-overview

### What part(s) of the page would you like to see updated?

There is an inconsistency with `filter` field indentation. 
In the top schema definition it's on the 1st level, but in some examples below `filter` is put into `type_params` at 2nd or 3rd level, what looks incorrect and fails at validation stage.  
Also it would be useful to emphasize that only multi-line YML format is supported by filter field and single-line won't work

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr updates the dimensions doc:
- add more info on valid_from and valid_to 
- adds implementation guidelines for scd ii tables
- updates the dimension name in the first code example.

help provided by @tlento [in slack convo](https://dbt-labs.slack.com/archives/C050Z6UP4RK/p1714413611285839?thread_ts=1714402934.686409&cid=C050Z6UP4RK)

Resolves #5395
Resolves #5396

### CODE REVIEW

File: website/docs/docs/build/dimensions.md [hunk 1]
```diff
@@ -202,30 +200,40 @@ measures:
...
-:::caution 
-Currently, there are limitations in supporting SCD's. 
+:::caution
+Currently, there are limitations in supporting SCD's.
 :::
 
-MetricFlow supports joins against dimensions values in a semantic model built on top of an SCD Type II table (slowly changing dimension) Type II table. This is useful when you need a particular metric sliced by a group that changes over time, such as the historical trends of sales by a customer's country. 
+MetricFlow supports joins against dimensions values in a semantic model built on top of an SCD Type II table (slowly changing dimension) Type II table. This is useful when you need a particular metric organized by a group that changes over time, such as the historical trends of sales by a customer's country.
+
+SCD Type II are groups that change values at a coarser time granularity. This results in a range of valid rows with different dimensions values for a given metric or measure. MetricFlow associates the metric with the first (minimum) available dimensions value within a coarser time window, such as month. By default, MetricFlow uses the group that is valid at the beginning of the time granularity.
```
Comment (mirnawong1) [inline website/docs/docs/build/dimensions.md]:
```suggestion
```

File: website/docs/docs/build/dimensions.md [hunk 2]
```diff
@@ -202,30 +200,39 @@ measures:
...
+:::caution
+Currently, there are limitations in supporting SCD's.
 :::
 
-MetricFlow supports joins against dimensions values in a semantic model built on top of an SCD Type II table (slowly changing dimension) Type II table. This is useful when you need a particular metric sliced by a group that changes over time, such as the historical trends of sales by a customer's country. 
+MetricFlow supports joins against dimensions values in a semantic model built on top of an SCD Type II table (slowly changing dimension) Type II table. This is useful when you need a particular metric organized by a group that changes over time, such as the historical trends of sales by a customer's country.
+
 
-As their name suggests SCD Type II are groups that change values at a coarser time granularity. This results in a range of valid rows with different dimensions values for a given metric or measure. MetricFlow associates the metric with the first (minimum) available dimensions value within a coarser time window, such as month. By default, MetricFlow uses the group that is valid at the beginning of the time granularity.
+**Basic structure**
```
Comment (mirnawong1) [inline website/docs/docs/build/dimensions.md:212]:
```suggestion
**Basic structure**

SCD Type II are groups that change values at a coarser time granularity. This results in a range of valid rows with different dimensions values for a given metric or measure. MetricFlow associates the metric with the first (minimum) available dimensions value within a coarser time window, such as month. By default, MetricFlow uses the group that is valid at the beginning of the time granularity.
```

File: website/docs/docs/build/dimensions.md [hunk 3]
```diff
@@ -202,29 +200,40 @@ measures:
...
 * `valid_from` (required): A timestamp indicating the start of a changing dimensions value must be present
 * `valid_to` (required): A timestamp indicating the end of a changing dimensions value must be present
 
-**Note**: The SCD dimensions table must have `valid_to` and `valid_from` columns.
+**Implementation**
 
-This is an example of SQL code that shows how a sample metric called `num_events` is joined with versioned dimensions data (stored in a table called `scd_dimensions`) using a primary key made up of the `entity_key` and `timestamp` columns. 
+Here are some guidelines to follow when implementing SCD Type II tables:
 
+- The SCD dimensions table must have `valid_to` and `valid_from` columns, which are logical constructs.
```
Comment (tlento) [inline website/docs/docs/build/dimensions.md]:
```suggestion
- The SCD semantic model must have `valid_to` and `valid_from` time dimensions, which are logical constructs.
```

File: website/docs/docs/build/dimensions.md [hunk 4]
```diff
@@ -202,29 +200,40 @@ measures:
...
+- The columns must be specified exactly once per dimension.
+- The columns shouldn't be used or specified on the same dimension.
+- The columns must cover a non-overlapping period where one row matches each natural key value (meaning they must not overlap and should be distinct).
```
Comment (tlento) [inline website/docs/docs/build/dimensions.md]:
```suggestion
- The `valid_from` and `valid_to` properties must be specified exactly once per SCD semantic model.
- The `valid_from` and `valid_to` properties shouldn't be used or specified on the same time dimension.
- The `valid_from` and 'valid_to` time dimensions must cover a non-overlapping period where one row matches each natural key value (meaning they must not overlap and should be distinct).
```

**Review by tlento — COMMENTED** [review]
This seems much more clear to me, thanks @mirnawong1 !

I left some suggestions for the SCD configuration. I think that might make it more obvious that the valid_from and valid_to constructs are purely semantic model annotations rather than something that has to be done in the database layer. If that deviates from how we describe these relationships elsewhere in the documents then please feel free to ignore it - the only important correction is that `valid_from` and `valid_to` must be specified once per semantic model.

**Comment by mirnawong1** [comment]
brilliant! thanks so much @tlento !! ✨

**Pull request merged**